tasks/Task.cpp

```diff
@@ -26,178 +26,128 @@ Task::~Task()
-    if( son == NULL )
-    {
-        printf("BVTSonar_Create: failed\n");
-        return 1;
-    }
-
-    // Open the sonar
-    int ret;
-    ret = BVTSonar_Open(son, "FILE", DataFile);
-    if( ret != 0 )
-    {
-        printf("BVTSonar_Open: ret=%d\n", ret);
-        return 1;
-    }
-
-    // Make sure we have the right number of heads
-    int heads = -1;
-    BVTSonar_GetHeadCount(son, &heads);
-    printf("BVTSonar_GetHeadCount: %d\n", heads);
-
-
-    // Get the first head
-    BVTHead head = NULL;
-    ret = BVTSonar_GetHead(son, 0, &head);
-    if( ret != 0 )
-    {
-        printf("BVTSonar_GetHead: ret=%d\n", ret);
-        return 1;
-    }
-
-    // Check the ping count
-    int pings = -1;
-    BVTHead_GetPingCount(head, &pings);
-    printf("BVTHead_GetPingCount: %d\n", pings);
-
-    // Check the min and max range in this file
-    float min_range, max_range;
-    BVTHead_GetMinimumRange(head, &min_range);
-    BVTHead_GetMaximumRange(head, &max_range);
-    printf("BVTHead_GetMinimumRange: %0.2f\n", min_range );
-    printf("BVTHead_GetMaximumRange: %0.2f\n", max_range );
-
-    BVTImageGenerator ig = BVTImageGenerator_Create();
-    BVTImageGenerator_SetHead(ig, head);
-
-    // Now, get a ping!
-    BVTPing ping = NULL;
-    ret = BVTHead_GetPing(head, 0, &ping);
-    if( ret != 0 )
+    try
     {
-        printf("BVTHead_GetPing: ret=%d\n", ret);
-        return 1;
-    }
+        // Open the sonar
+        std::string ip_address = _ip_address.get();
+        son.Open("NET", _ip_address.get().c_str());
 
-    // Generate an image from the ping
-    BVTMagImage img;
-    ret = BVTImageGenerator_GetImageXY(ig, ping, &img);
-    if( ret != 0 )
-    {
-        printf("BVTImageGenerator_GetImageXY: ret=%d\n", ret);
-        return 1;
-    }
+        head = son.GetHead(0);
 
-    printf("\n");
+        //set the range
+        this->setRange(_range.get());
+        //set the analog gain adjustment (in dB)
+        head.SetGainAdjustment(_gain.get());
 
-    /////////////////////////////////////////////////////////
+        //set Clipping Threshold
+        head.SetClippingThreshold(_clipping_thresh.get());
 
-    // Check the image height and width out
-    int height;
-    BVTMagImage_GetHeight(img, &height);
-    printf("BVTMagImage_GetHeight: %d\n", height);
-    int width;
-    BVTMagImage_GetWidth(img, &width);
-    printf("BVTMagImage_GetWidth: %d\n", width);
+        //set TVG Slope
+        head.SetTVGSlope(_tvg_slope.get());
 
-    // Save it to a PGM (PortableGreyMap)
-    ret = BVTMagImage_SavePGM(img, "img.pgm");
-    if( ret != 0 )
-    {
-        printf("BVTMagImage_SavePGM: ret=%d\n", ret);
-        return 1;
-    }
+        //set the speed of sound
+        head.SetSoundSpeed(_sound_speed.get());
 
-    /////////////////////////////////////////////////////////
+        //set Fluid Type
+        this->setFluidType(_fluid_type.get());
 
-    // Build a color mapper
-    BVTColorMapper mapper;
```

Comment:
Why do you do color mapping in here ? Don't do any visualization inside the component. Ever. Use Romulo's sonar visualization from outside it.